tools/installPluginVersion.php

```diff
@@ -55,6 +55,43 @@ function execute() {
 		$pluginVersion = $versionInfo['version'];
 		$versionDao = DAORegistry::getDAO('VersionDAO');
 		$versionDao->insertVersion($pluginVersion, true);
+
+		$pluginPath = dirname($this->_descriptor);
+		$pluginCategory = preg_replace('/^plugins\./', '', $pluginVersion->getProductType());
```

Comment:
I was thinking `preg_match`, rather than `preg_replace`, and there's already a whitelist available in `Application::getPluginCategories`.